src/source/tile.js

```diff
@@ -19,11 +20,44 @@ const CLOCK_SKEW_RETRY_TIMEOUT = 30000;
  * @private
  */
 class Tile {
+    coord: any;
+    uid: number;
+    uses: number;
+    tileSize: number;
+    sourceMaxZoom: number;
+    buckets: any;
+    expirationTime: any;
+    expiredRequestCount: number;
+    state: 'loading'   // Tile data is in the process of loading.
+         | 'loaded'    // Tile data has been loaded. Tile can be rendered.
+         | 'reloading' // Tile data has been loaded and is being updated. Tile can be rendered.
+         | 'unloaded'  // Tile data has been deleted.
+         | 'errored'   // Tile data was not loaded because of an error.
+         | 'expired';  // Tile data was previously loaded, but has expired per its
+                       // HTTP headers and is in the process of refreshing.
+    placementThrottler: any;
+    timeAdded: any;
+    fadeEndTime: any;
+    rawTileData: any;
+    collisionBoxArray: any;
+    collisionTile: ?CollisionTile;
+    featureIndex: any;
+    redoWhenDone: boolean;
+    angle: number;
+    pitch: number;
+    cameraToCenterDistance: number;
+    cameraToTileDistance: number;
+    showCollisionBoxes: boolean;
+    placementSource: any;
+    workerID: number;
+    vtLayers: any;
+
     /**
      * @param {TileCoord} coord
-     * @param {number} size
+     * @param size
+     * @param sourceMaxZoom
      */
-    constructor(coord, size, sourceMaxZoom) {
+    constructor(coord: any, size: number, sourceMaxZoom: number) {
```

Comment:
This was a rough pass; for many types I just used `any` where flow would otherwise complain. `tile_coord.js` hasn't been typed yet.